cachito/workers/tasks/pip.py

```diff
@@ -108,3 +119,55 @@ def fetch_pip_source(request_id, package_configs=None):
 
     if pip_config_files:
         update_request_with_config_files(request_id, pip_config_files)
+
+
+def _get_custom_requirement_config_file(requirement_file_path, source_dir, raw_repo_name):
+    """
+    Get custom pip requirement file.
+
+    Generates and returns a configuration file representing a custom pip requirement file where the
+    original URL and VCS entries are replaced with entries pointing to those resources in Nexus.
+
+    :param str requirement_file_path: path to the requirement file
+    :param Path source_dir: path to the application source code
+    :param str raw_repo_name: name of the raw hosted Nexus repository containing the
+        requirements
+    :return: Cachito configuration file representation containing the custom requirement file
+    :rtype: dict
+    """
+    original_requirement_file = PipRequirementsFile(requirement_file_path)
+    cachito_requirements = []
+    differs_from_original = False
+    for requirement in original_requirement_file.requirements:
+        raw_component_name = get_raw_component_name(requirement)
+        if raw_component_name:
+            # increase max_attempts to make sure the package upload/setup is complete
+            new_url = nexus.get_raw_component_asset_url(
```

Comment:
I'm wondering if we need to include some of the parts from the original URL, in case it contains, for example, the `#egg` fragment. We may need to add the `fragment` part, and possibly also the `query` part of the original url. Not sure about the `params` part (I don't know what those are)

Reply:
Nevermind, the `copy()` method already takes care of  this.